docs/src/content/docs/book/functions.mdx

```diff
@@ -114,6 +116,175 @@ native storeUint(s: Builder, value: Int, bits: Int): Builder;
 extends mutates native loadInt(self: Slice, l: Int): Int;
 ```
 
+## Assembly functions, `asm` {#asm}
+
+<Badge text="Available since Tact 1.5" variant="tip" size="medium"/><p/>
+
+:::caution
+
+  These are very advanced functions that require experience and vigilance in both definitions and usage. The logical errors in them are extremely hard to spot, the error messages are abysmal, and type checking isn't currently provided by Tact.
+
+  That said, if you know what you're doing, they can offer you the smallest possible gas usage, the best performance and the most control over [TVM][tvm] execution. Remember — with great power comes great responsibility.
+
+:::
+
+Assembly functions (or `asm{:tact}` functions for short) are module-level functions that allow writing [Fift and TVM assembly](https://docs.ton.org/v3/documentation/smart-contracts/fift/fift-and-tvm-assembly) directly in Tact. Unlike all other functions, their bodies consist only of [Fift words][fift] and [TVM instructions][tvm-instructions], and don't use any [Tact statements](/book/statements).
+
+```tact
+// all assembly functions must start with "asm" keyword
+// ↓
+   asm fun answer(): Int { 42 INT }
+//                         ------
+//                         Notice, that the body contains
+//                         only Fift words or TVM instructions
+```
+
+### Caveats {#asm-caveats}
+
+The difference between Fift words and TVM instructions is that words are processed and computed at [compile-time](/ref/core-comptime), while instructions are converted to opcodes and embedded into the contract as part of its code. They are also visually different: TVM instructions are almost all uppercase.
+
+```tact
+/// Defines ++ word to be used elsewhere
```

Comment:
definitely no `++` should be officially supported

Reply:
Will be removed with the whole Fift section, other caveats are to be adjusted for TVM instructions and questions from @jeshecdom